### DISCUSSION TITLE
Using Birdnet as a source for OBS / Live streaming

### DISCUSSION THREAD

**Discussion by London-Lass** [mcguirepr89/BirdNET-Pi#792] [General]
I'm considering the possibility that birdnet might be able to supply content to a livestream (via OBS). I understand that OBS can use either a media source, or indeed a stream (rtsp?). Ideal candidates if they could be streamed via rtsp :-

*  spectogram
* today's detections
* species stats

I understand it would be trivial to grab the individual images via automation and then convert from image to video, but i'm hoping that content such as spectrogram might be accessible easliy in a more direct format.

**Comment by ehpersonal38** [comment]
Unfortunately the live spectrogram, Today's Detections, and Species Stats all can only run in a browser. So I'm not sure what you could do other than have OBS capture a browser window with those pages open. 

However, if you want an audio stream, you can use the Icecast2 stream which can be accessed at the following URL of your BirdNET-Pi:
`birdnet:{your birdnet-pi password}@birdnetpi.local/stream`

For example, if my password is "ilovebirds", the full URL will be:
`birdnet:ilovebirds@birdnetpi.local/stream`

You can also replace `birdnetpi.local` with your Pi's IP address and all will be the same.

**Comment by fherb2** [comment]
Hi,

Since the site where my microphone is working has a small internet connection, and I want to use the sound for secondary purposes, my setup works following:

RaspberryPi withSoundCard and simple-rtsp-server ---> packed into a VPN tunnle ---> __LTE-connection__ ---> unpacking VPN at my home ---> (*) Raspberry Pi with  simple-rtspr-server as proxy ---> BirdNetPi software at the same Raspberry Pi

and additional starting with "(*) Raspberry Pi with  simple-rtspr-server as proxy":

... (*) Raspberry Pi with  simple-rtspr-server as proxy ---> hosted Linux server at Netcup with simple-rtspr-server as proxy ---> to anybody like Locus Sonus ...

So the Netcup server with a good Ethernet connection of 1GBit/s is the distributor of this stream to anybody.

I don't know what OBS needs as input. But a RTSP delivery (and some other formats) are possible by using the simple-rtsp-server from: https://github.com/aler9/rtsp-simple-server

I'm using the directly installation at my sound source RaspberryPi with the sound card and the microphones. All other stations uses the docker image.

Maybe this ideas are helping you.

You can try this setup at https://www.karlincam.cz/de_de/sound-of : -> Live-Sound aus Karlin: ...
Sound comes from my Netcup server.

Frank

**Reply by ampsu** [reply]
That approach looks fun. I wonder, does streaming audio from the location use less hardware resources than analyzing on site? Lets say, if I run the setup on a battery, can lower power Pi Zero handle the audio stream? I am looking into every solution to maximize my battery powered Pi stations. Maybe just having multiple audio streams is better than having multiple Pi-s analyzing.

**Reply by ehpersonal38** [reply]
@ampsu You could try a little experiment. Disable all services except Livestream service, and let your Pi do it's thing for 15 minutes, then take note of its 15-min load average using a tool like `htop`. Then repeat the experiment with Recording Service, BirdNET client, and BirdNET server all running and the rest disabled. Let that go for 15 mins, and see which one uses more load.

**Reply by fherb2** [reply]
Some words, some weeks later:

My configuration works really good and stable. Also in times where the internet connection is not really good regarding bandwidth. I convert the stereo audio signal to MP3 with 160kBit/s. So the upload traffic is in the middle about 21...22kByte/s. 160kBit/s are  20kByte/s. With the offset of wireguard encryption and TCP/IP overhead, 21...22kByte/s are typically.

The used RTSP server and protocol is a good solution to send my stream to my BirdNetPi inclusive via relay at BirdNetPi Raspberry Pi to my Netcup server. But is is not a good solution to send an audio stream from there to users, Locu Sonus and (since some days) birdweather.com by using RTSP. Reason is a bad control interface between server and client for good buffering the data before converting to the audio signal. A lot of brakes. Not nice to hear. And most browsers can not play MP3 or other formats from an RTSP containerized stream.

I solved this problem by changing the RTSP server at my public Netcup server against an Icecast2 Server. This is not only compatible to Locu Sonus and Birdweather (my station here: https://app.birdweather.com/stations/1036). It's also really better to send the electronic sound to a client for real-time converting into acoustic sound. There are less breaks sometimes. But the control channel of Icecast2 between server and client is really good. If the internet connection is not really a big bottleneck, so the sound is seldom breaking for buffering.

An additional information: I checked to use Ogg Opus / 128kBit/s instead MP3 / 160kBit/s. The sound quality is very good. But it is a problem to receive this format at Apple devices. So I stay with MP3/160kBit/s what is at least the same quality. Bandwidth via my VPN tunnel would decrease from 21...22kByte/s to about 17kByte/s with Ogg Opus 128kbit/s. But, ok: Some more data rate but better compatibility.

Best regards, Frank
